src/session/index.ts

```diff
@@ -242,6 +243,19 @@ export default abstract class Session extends EventEmitter {
         return taskScript;
     }
 
+    getProxyUrl (url: string): string {
+        if (!this.proxy?.server2Info)
```

Comment:
I guess it should be more complicated:
* need to determine the url type. If it's same-domain, the use the `server1Info`, else - `server2Info`.
* before creating proxy url use this [preparation](https://github.com/DevExpress/testcafe-hammerhead/blob/f4364fb67e5c588b8033b7183a318a1615fee17c/src/proxy/index.ts#L238).

The using other options (`externalProxySettings`, `disableHttp2`) we will consider later.